components/tidb_query_expr/src/impl_string.rs

```diff
@@ -3124,26 +3137,68 @@ mod tests {
     #[test]
     fn test_strcmp() {
         let test_cases = vec![
-            (Some(b"123".to_vec()), Some(b"123".to_vec()), Some(0)),
-            (Some(b"123".to_vec()), Some(b"1".to_vec()), Some(1)),
-            (Some(b"1".to_vec()), Some(b"123".to_vec()), Some(-1)),
-            (Some(b"123".to_vec()), Some(b"45".to_vec()), Some(-1)),
+            (
+                Some(b"123".to_vec()),
+                Some(b"123".to_vec()),
```

Comment:
Please add a test case for `ABC`, `abc` with a CASE_INSENSITIVE collator.

Reply:
> Sign-off all commits requires a force-puch I guess.

Done, PTAL

Reply:
Done